### ISSUE TITLE
Feature Request: The ability to overlay segmentation masks on images

### ISSUE THREAD

**Issue by NewtonsLadle** [tensorflow/tensorboard#1042]
For semantic segmentation work, it would be nice to be able to specify image backgrounds and have the serialized data overlay those images as colored masks, maybe with variable opacity.

**Comment by chihuahua** [comment]
I think this feature could make for a useful [TensorBoard plugin](https://github.com/tensorflow/tensorboard-plugin-example/blob/master/README.md), which extends TensorBoard's current capabilities with visualizations.

What would you think about an API that is used like this?

```python
tb.summary.overlay(image, overlay)
```

where `overlay` is an image (containing boxes for overlay) that is the same shape as `image`. What kind of API would you find useful?

**Comment by NewtonsLadle** [comment]
I agree that it would make a good plugin. What do you mean by "boxes for overlay"? 

I think that being able to specify the opacity of the overlay is important, but maybe that should be controlled on the client side. I'd also like it to be treated as images are where you can look at the history of the overlay with a slider.
